ckanext/googleanalytics/plugin/pylons_plugin.py

```diff
@@ -0,0 +1,181 @@
+            m.connect(
+                "/action/{logic_function}",
+                action="action",
+                conditions=GET_POST,
+            )
+
+        # /api ver 1, 2, 3 or none
+        with SubMapper(
+            map,
+            controller="ckanext.googleanalytics.controller:GAApiController",
+            path_prefix="/api{ver:/1|/2|/3|}",
+            ver="/1",
+        ) as m:
+            m.connect("/search/{register}", action="search")
+
+        # /api/rest ver 1, 2 or none
+        with SubMapper(
+            map,
+            controller="ckanext.googleanalytics.controller:GAApiController",
+            path_prefix="/api{ver:/1|/2|}",
+            ver="/1",
+            requirements=dict(register=register_list_str),
+        ) as m:
+
+            m.connect("/rest/{register}", action="list", conditions=GET)
+            m.connect("/rest/{register}", action="create", conditions=POST)
+            m.connect("/rest/{register}/{id}", action="show", conditions=GET)
+            m.connect("/rest/{register}/{id}", action="update", conditions=PUT)
+            m.connect(
+                "/rest/{register}/{id}", action="update", conditions=POST
+            )
+            m.connect(
+                "/rest/{register}/{id}", action="delete", conditions=DELETE
+            )
+
+        return map
+
+    def after_map(self, map):
+        """Add new routes that this extension's controllers handle.
+
+        See IRoutes.
+
+        """
+        self._modify_resource_download_route(map)
+        map.redirect("/analytics/package/top", "/analytics/dataset/top")
+        map.connect(
+            "analytics",
+            "/analytics/dataset/top",
+            controller="ckanext.googleanalytics.controller:GAController",
+            action="view",
+        )
+        return map
+
+    def _modify_resource_download_route(self, map):
+        """Modifies resource_download method in related controller
+        to attach GA tracking code.
+        """
+
+        if "_routenames" in map.__dict__:
+            if "resource_download" in map.__dict__["_routenames"]:
+                route_data = map.__dict__["_routenames"][
+                    "resource_download"
+                ].__dict__
+                route_controller = route_data["defaults"]["controller"].split(
+                    ":"
+                )
+                module = importlib.import_module(route_controller[0])
+                controller_class = getattr(module, route_controller[1])
+                controller_class.resource_download = wrap_resource_download(
+                    controller_class.resource_download
+                )
+            else:
+                # If no custom uploader applied, use the default one
+                PackageController.resource_download = wrap_resource_download(
+                    PackageController.resource_download
+                )
+
+
+def wrap_resource_download(func):
+    def func_wrapper(cls, id, resource_id, filename=None):
```

Comment:
There should be wrappers for sending multiple analytics (analytics by resource, package, and organization)
https://github.com/OpenGov-OpenData/ckanext-googleanalytics/blob/master/ckanext/googleanalytics/plugin.py#L64-L113